src/main/java/eu/dissco/core/translator/service/BioCaseService.java

```diff
@@ -171,76 +180,81 @@ private void mapABCD206(XMLEventReader xmlEventReader)
 
   private void processUnit(DataSet datasets, Unit unit)
       throws JsonProcessingException, DisscoEfgParsingException {
-    var unitData = getData(mapper.valueToTree(unit));
-    extractEfgInformation(unit, unitData);
-    if (isAcceptedType(unit)) {
-      var organizationId = getProperty("organization_id", unitData);
-      var physicalSpecimenIdType = getProperty("physical_specimen_id_type", unitData);
+    var unitAttributes = parseToJson(unit);
+    if (isAcceptedBasisOfRecord(unit)) {
+      var organizationId = termMapper.retrieveFromABCD(new OrganisationId(), unitAttributes);
+      var physicalSpecimenIdType = termMapper.retrieveFromABCD(new PhysicalSpecimenIdType(),
+          unitAttributes);
       if (physicalSpecimenIdType != null) {
         var physicalSpecimenId = getPhysicalSpecimenId(physicalSpecimenIdType, organizationId,
-            unitData);
+            unitAttributes);
         var digitalSpecimen = new DigitalSpecimen(
-            getProperty("type", unitData),
             physicalSpecimenId,
-            physicalSpecimenIdType,
-            getProperty("physical_specimen_collection", unitData),
-            getSpecimenName(unitData),
-            organizationId,
-            datasets.getDatasetGUID(),
-            webClientProperties.getSourceSystemId(),
-            removeMappedFields(unitData),
-            unitData,
-            null
+            termMapper.retrieveFromABCD(new Type(), unitAttributes),
+            harmonizeAttributes(datasets, unitAttributes, physicalSpecimenIdType, organizationId),
+            unitAttributes
         );
+
         log.debug("Result digital Specimen: {}", digitalSpecimen);
```

Comment:
Do we want to make a kind of guideline for creating logs within the DiSSCo environment? To apply to the other back- and front-end applications (UCAS, ELViS, etc.)

Reply:
Not sure, never had a guideline. I think the general rule is do logging, enough to trace what is going on but not so much you can't find anything

Reply:
Created this ticket https://naturalis.atlassian.net/browse/DD-190